projects/plugins/social/src/class-note.php

```diff
@@ -106,4 +127,26 @@ public function toggle_enabled_status() {
 		// Delete this option, so the rules get flushe in maybe_flush_rewrite_rules when the CPT is registered.
 		delete_option( self::FLUSH_REWRITE_RULES_FLUSHED );
 	}
+
+	/**
+	 * Use the_title hook so we show the social note's exceprt in the post list view.
+	 *
+	 * @param array $title The title of the post, which we have set to be an empty string for Social Notes.
+	 * @param array $post_id The Post ID.
+	 */
+	public function override_empty_title( $title, $post_id ) {
+		if ( get_post_type( $post_id ) === self::JETPACK_SOCIAL_NOTE_CPT ) {
+			$words = explode( ' ', wp_strip_all_tags( wp_trim_excerpt( get_post_field( 'post_content', $post_id ) ) ) );
+
+			// Get the first 10 words
+			$first_10_words = array_slice( $words, 0, 10 );
+
+			// Join the first 10 words back into a string
+			return implode( ' ', $first_10_words );
+
+		}
```

Comment:
I think you were using `wp_trim_excerpt` before. We can control the excerpt length via filter with that, so an alternative for this would be something like:

```suggestion
			$excerpt_filter = function ( $length ) { return 10; };
			add_filter( 'excerpt_length', $excerpt_filter );
			$title = wp_trim_excerpt( '', $post_id );
			remove_filter( 'excerpt_length, $excerpt_filter );
			return $title;
		}
```

I'm not sure which is better, but it might be worth us trying to use the word break logic from core if we can.

Reply:
I have gone a slightly different way with this without the filter, let me know what you think! @pablinos

Reply:
Nice, that works too!